tests/osgi-tests/src/test/java/org/wso2/carbon/osgi/testcontainer/DistributionDirectoryTest.java

```diff
@@ -0,0 +1,54 @@
+package org.wso2.carbon.osgi.testcontainer;
+
+import org.ops4j.pax.exam.Configuration;
+import org.ops4j.pax.exam.ExamFactory;
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.ops4j.pax.exam.testng.listener.PaxExam;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.testng.Assert;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+import org.wso2.carbon.container.CarbonContainerFactory;
+import org.wso2.carbon.kernel.utils.CarbonServerInfo;
+
+import java.nio.file.Paths;
+import javax.inject.Inject;
+
+import static org.wso2.carbon.container.options.CarbonDistributionOption.carbonDistribution;
+
+/**
+ * To test the pax exam container using distribution directory.
+ */
+@Listeners(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+@ExamFactory(CarbonContainerFactory.class)
+public class DistributionDirectoryTest {
+
+    @Inject
+    protected BundleContext bundleContext;
+
+    @Inject
+    private CarbonServerInfo carbonServerInfo;
+
+    @Configuration
+    public Option[] config() {
+        return new Option[] { carbonDistribution(Paths.get("target", "wso2carbon-kernel-test-5.2.0-SNAPSHOT")), };
```

Comment:
Don't use 5.2.0-SNAPSHOT, this will break the test case in next release,
Instead derive it programmatically.